compose/sbcp/sbcp.go

```diff
@@ -0,0 +1,39 @@
+package sbcp
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+
+	"github.com/compose-network/specs/compose"
+)
+
+// GenerateInstanceID returns SHA256(periodID || seq || tx1 || tx2 || ... || txn)
+func GenerateInstanceID(
+	periodID compose.PeriodID,
+	seq compose.SequenceNumber,
+	xtRequest compose.XTRequest,
+) compose.InstanceID {
+	var b [8]byte
+	buf := bytes.NewBuffer(nil)
+
+	// Encode period timestamp (nanoseconds) as 8 bytes big-endian
+	binary.BigEndian.PutUint64(b[:], uint64(periodID))
+	buf.Write(b[:])
+
+	// Encode sequence number as 8 bytes big-endian
+	binary.BigEndian.PutUint64(b[:], uint64(seq))
+	buf.Write(b[:])
+
+	// Append each transaction's raw bytes
+	for _, req := range xtRequest.Transactions {
+		for _, data := range req.Transactions {
+			if len(data) > 0 {
+				buf.Write(data)
+			}
+		}
+	}
+
```

Comment:
We need to better handle collisions.
1. Hash ChainIDs as well. Transactions replayed across different chains shouldn't yield the same instanceID.
2. Hash tx length as well. Note  `txA=[0x01], txB=[0x02]` and `txSingle=[0x01,0x02]` produce the same byte stream `(0x01 0x02)` and therefore the same hash. Something similar may happen with a carefully planned attack? Better stay on the safe side!

Observation: simply hashing the proto serailization of `XTRequest` will automatically embed it all, but then InstanceID may change due to changes in `protoc` lib (not sure this is an issue, but something stable is always better)